iam/src/providers/githubStarredRepoProvider.ts

```diff
@@ -0,0 +1,151 @@
+// ----- Types
+import type { RequestPayload, VerifiedPayload } from "@gitcoin/passport-types";
+import type { Provider, ProviderOptions } from "../types";
+import type {
+  GithubFindMyUserResponse,
+  GithubRepoRequestResponse,
+  GithubUserRepoResponseData,
+} from "./types/githubTypes";
+
+// ----- HTTP Client
+import axios from "axios";
+
+export type GithubTokenResponse = {
+  access_token: string;
+};
+
+// Export a Github Provider to carry out OAuth and return a record object
+export class StarredGithubRepoProvider implements Provider {
+  // Give the provider a type so that we can select it with a payload
+  type = "StarredGithubRepoProvider";
+
+  // Options can be set here and/or via the constructor
+  _options = {};
+
+  // construct the provider instance with supplied options
+  constructor(options: ProviderOptions = {}) {
+    this._options = { ...this._options, ...options };
+  }
+
+  // verify that the proof object contains valid === "true"
+  async verify(payload: RequestPayload): Promise<VerifiedPayload> {
+    let valid = false,
+      verifiedUserPayload: GithubFindMyUserResponse = {},
+      verifiedUserRepoPayload: GithubUserRepoResponseData = {};
+
+    try {
+      verifiedUserPayload = await verifyGithub(payload.proofs.code);
+      verifiedUserRepoPayload = await verifyUserGithubRepo(verifiedUserPayload, payload.proofs.code);
```

Comment:
Same observation as above: get the access token once, and pass it in as argument to both `verify*` functions